src/map.rs

```diff
@@ -3262,7 +3264,16 @@ mod test_map {
         }
 
         let mut m = HashMap::new();
+
+        // FIXME: https://github.com/rust-lang/miri/issues/653
+        #[cfg(not(miri))]
         let mut rng = thread_rng();
+        #[cfg(miri)]
+        let mut rng = {
+            use rand::SeedableRng;
+            let seed = [1,2,3,4, 5,6,7,8, 9,10,11,12, 13,14,15,16];
+            rand::rngs::SmallRng::from_seed(seed)
+        };
```

Comment:
I personally feel this would look nicer with `if cfg!(miri)`.

Also run cargo fmt.

Reply:
> Also run cargo fmt.

It's not available in the last nightly :/

Reply:
I've reformatted, rustfmt failed a couple of hours ago to install for me, but I just retried and it just worked =/

Reply:
>  personally feel this would look nicer with if cfg!(miri).

Note that the type of `rng` differs, one is the thread local rng, and the other one is a SmallRng, so without conditional compilation type checking fails.

Reply:
On second thought we might as well simply always make that test use `SmallRng`.

Reply:
Note that `SmallRng` with a non-random seed will always generate the exact same sequence of elements (which is why it works on miri).

Reply:
That's fine. The test doesn't need true randomness anyways.